docs/RFCs/package-toml.md

```diff
@@ -0,0 +1,128 @@
+1. The `package.json` name specifier does not allow for the namespacing that entropic provides
+1. There are recognized deficiencies with using JSON as human-readable metadata format
+
+## Filename
+
+The filename of the manifest **MUST** be `Package.toml` and it **MUST** live in
+the root directory of the package's filesystem. The root of the package is defined
+as the closest parent directory that contains the manifest file, starting with
+the current directory.
+
+On case-insensitive filesystems, it **MAY** be presented as `package.toml` but care
+should be taken to preserve case when possible.
+
+## Format
+
+The format of this file is **REQUIRED** to be in [TOML](https://github.com/toml-lang/toml)
+version 0.5.0 or later as defined by the project.
+
+### Top-Level Data
+
+The manifest **MUST** contain the following top-level keys:
+
+- `"name"` - the canonical name of the package, as a string, including it's namespace. e.g. `"toddself@static-pkg.dev/packge2toml"`
+- `"version"` - a [SEMVER](https://semver.org/) designation conforming to version 2.0.0 of the semantic versioning specification. Due to how TOML parses numbers, this must be represented as a string. e.g. `"1.0.4"`
+
+The manifest **MAY** contain the following fields:
+
+- `"entry"` - the filename and package-relative path, as a string, to the main entry point of the package. The root for the relative-path is the directory which contains the `Package.toml` file. e.g `"./src/index.js"`
+- `"type"` - if set to `"module"` this will output a [`package.json` with this included](https://nodejs.org/api/esm.html#esm_code_package_json_code_code_type_code_field), and will signal to the node interpreter to run this code an ES6 Module.
+- `"license"` - a valid [SPDX license](https://spdx.org/licenses/) for your code. If a license is not provided then the code is considered to be `UNLICENSED`.
+- `"description"` - a short description of your package
+- `"homepage"` - a URL for your project's homepage
+- `"author"` - who wrote this package
+- `"repository"` - where does this code live
+
+These fields **MAY** or **MAY NOT** be validated by CLI tools or the server-side
+software, depending on the implementation.
+
+Software implementing the entropic federation protocols and `Package.toml` handling **MAY** complain about any of these optional fields as missing (`license` being an obvious candidate).
+
+### Subsections
+
+The manifest file **MAY** contain zero or more subsections as defined below.
+
+#### Dependency Lists
+
+Dependencies lists contain all the resources that are required for
+this package, in various scenarios. The dependencies **MUST** be declared as `"key" = "value"`
+pairs where the `key` is the package's canonical name (as from the top-level section
+of it's `Package.toml` file) and the `value` is a valid SEMVER version or range
+expression that points to a valid package.
+
+Examples:
+
+To specify a specific version:
+
+```
+"toddself@static-pkg.dev/package2toml" = "1.0.4"
```

Comment:
A thought comes to mind that these left-hand-side package names are _incredibly unwieldy_, and we'd be better served with leaving these long specifiers to the right-hand side. For example
```toml
[dependencies]
package2toml = "toddself@static-pkg.dev/package2toml@1.0.4"
```
This would instruct the package manager to make the name available as `package2toml` instead of `LONG@NAME/HERE`.

To riff off my earlier comment about objects, users might also choose to split these sections up:

```toml
[dependencies]
package2toml = { from = "toddself@static-pkg.dev/package2toml", version: "1.0.4" }
```

Reply:
another thought:

```toml
"toddself@static-pkg.dev/package2toml" = { version: "1.0.4", alias: "package2toml" }
```

but tbh I like the first example I gave much better.

Reply:
Oh and from an implementation point of view, having a full-fledged specifier on the RHS makes it much easier to have this sort of thing coexist with existing specifiers (as in, https://github.com/npm/npm-package-arg)

Reply:
(We've discussing whether or not we want to support github-style deps – right now my M.O. is to only allow git deps when run at application level, for patching dependencies.)

Reply:
The long names are a bit awkward, but doesn't assigning them to an arbitrary alias lead to potential weirdness with transitive dependencies? If I make lodash `_` and someone else makes it `lodash`, what happens when someone depends on both our libraries? Letting dependencies dictate what they're called seems like a sensible standard.

Reply:
@chrisdickinson I like what Cargo does on this front: you're not allowed to publish using git deps but they're fine for local usage. Also yes, a patch feature would be nice.

@dmfay the package manager is able to differentiate those based on their full identifier + aliases. This is how the alias feature currently works in npm. I'm not worried about conflicts. The only downside here is that there might be less deduplication in the tree.

Reply:
I hadn't heard of an alias feature in npm! I didn't turn anything up in the docs, but I found [this](https://github.com/npm/npm/issues/2943) on the issue tracker which seems to be a conclusive 'no'. Could you elaborate?

I do think there's more at stake than duplication: it becomes less likely that copy+pasted code will "just work" if dependencies it references don't have to have a consistent name, which sounds more annoying the longer I think about it. And I don't know how it'd play with Symbols.

Reply:
Sorry, it's my bad. Aliases are... not really documented, and it's a fairly recent feature.

tl;dr you can do:

```json
"dependencies": {
  "underscore": "npm:lodash@1.2.3"
}
```

and it'll install `lodash` under `node_modules/underscore`.